integration/tests/cook/cli.py

```diff
@@ -25,18 +26,28 @@ def stdout(cp):
     return decode(cp.stdout).strip()
 
 
-def sh(command, stdin=None, env=None):
+def sh(command, stdin=None, env=None, wait_for_exit=True):
     """Runs command using subprocess.run"""
     logger.info(command + ((' # stdin: %s' % decode(stdin)) if stdin else ''))
-    cp = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, input=stdin, env=env)
-    return cp
-
-
-def cli(args, cook_url=None, flags=None, stdin=None, env=None):
+    command_args = shlex.split(command)
+    if wait_for_exit:
+        cp = subprocess.run(command_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, input=stdin, env=env)
+        return cp
+    else:
+        proc = subprocess.Popen(command_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        # Get the current stdout flags
+        flags = fcntl(proc.stdout, F_GETFL)
+        # Set the O_NONBLOCK flag of the stdout file descriptor
+        # (if we don't set this, calls to readlines() will block)
+        fcntl(proc.stdout, F_SETFL, flags | os.O_NONBLOCK)
+        return proc
+
+
+def cli(args, cook_url=None, flags=None, stdin=None, env=None, wait_for_exit=True):
```

Comment:
f-strings for all format-strings in this function?

Reply:
Done